pkg/plugins/policies/meshtcproute/plugin/v1alpha1/listeners_backendrefs.go

```diff
@@ -45,12 +17,34 @@ func getBackendRefs(
 ) []common_api.BackendRef {
 	service := core_xds.MeshService(serviceName)
 
+	tcpConf := core_xds.ComputeConf[api.Rule](toRulesTCP, service)
+
 	// If the outbounds protocol is http-like and there exists MeshHTTPRoute
 	// with rule targeting the same MeshService as MeshTCPRoute, it should take
 	// precedence over the latter
-	if matchingHTTPRuleExist(toRulesHTTP, service, protocol) {
-		return nil
+	switch protocol {
+	case core_mesh.ProtocolHTTP, core_mesh.ProtocolHTTP2, core_mesh.ProtocolGRPC:
+		// If we have an >= HTTP service, don't manage routing with
+		// MeshTCPRoutes if we either don't have any MeshTCPRoutes or we have
+		// MeshHTTPRoutes
+		httpConf := core_xds.ComputeConf[meshhttproute_api.PolicyDefault](
+			toRulesHTTP,
+			service,
+		)
+		if tcpConf == nil || httpConf != nil {
+			return nil
+		}
+	default:
 	}
 
-	return getTCPBackendRefs(toRulesTCP, service)
+	if tcpConf != nil {
+		return tcpConf.Default.BackendRefs
+	}
+	return []common_api.BackendRef{{
+		TargetRef: common_api.TargetRef{
+			Kind: common_api.MeshService,
+			Name: serviceName,
+		},
+		Weight: pointer.To(uint(100)),
```

Comment:
Shouldn't be `1`?

Reply:
I think it doesn't matter since it's the only backend

Reply:
Seems like there is a nil pointer https://github.com/kumahq/kuma/blob/master/pkg/plugins/policies/meshhttproute/plugin/v1alpha1/gateway.go#L37